src/System.Device.Gpio/System/Device/Gpio/Drivers/LibgpiodDriver.cs

```diff
@@ -0,0 +1,195 @@
+            return PinMode.Input;
+        }
+
+        protected internal override bool IsPinModeSupported(int pinNumber, PinMode mode)
+        {
+            // Libgpiod command line tools does not support pull up or pull down resistors.
+            return mode != PinMode.InputPullDown && mode != PinMode.InputPullUp;
+        }
+
+        protected internal override void OpenPin(int pinNumber)
+        {
+            // do nothing
+        }
+
+        protected internal override PinValue Read(int pinNumber)
+        {
+            PinValue result = default(PinValue);
+            Process gpioGetCommand = new Process();
+            gpioGetCommand.StartInfo.FileName = GpioGet;
+            gpioGetCommand.StartInfo.Arguments = $"{GpioChip0} {pinNumber}";
+            gpioGetCommand.StartInfo.RedirectStandardOutput = true;
+            gpioGetCommand.Start();
+            if (WaitForProcessToExit(gpioGetCommand, $"{GpioGet} {GpioChip0} {pinNumber}", TimeoutMilliSeconds))
+            {
+                string valueContents = gpioGetCommand.StandardOutput.ReadToEnd();
+                result = ConvertMachineValueToPinValue(valueContents);
+            }
+            return result;
+        }
+
+        private PinValue ConvertMachineValueToPinValue(string value)
+        {
+            return value != null && value.IndexOf('1') != -1 ? PinValue.High : PinValue.Low;
+        }
+
+        protected internal override void RemoveCallbackForPinValueChangedEvent(int pinNumber, PinChangeEventHandler callback)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected internal override void SetPinMode(int pinNumber, PinMode mode)
+        {
+            if (mode == PinMode.InputPullDown || mode == PinMode.InputPullUp)
+            {
+                throw new PlatformNotSupportedException("This driver is generic so it does not support Input Pull Down or Input Pull Up modes.");
+            }
+            MutableTuple<PinMode, Process> pinModeAndProcess;
+            if (_pinNumberToPinModeAndProcess.TryGetValue(pinNumber, out pinModeAndProcess))
+            {
+                pinModeAndProcess.Value1 = mode;
+            }
+            else
+            {
+                _pinNumberToPinModeAndProcess.Add(pinNumber, new MutableTuple<PinMode, Process>(mode, null));
+            }
+        }
+
+        protected internal override WaitForEventResult WaitForEvent(int pinNumber, PinEventTypes eventType, CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();  
+        }
+
+        protected internal override void Write(int pinNumber, PinValue value)
+        {
+                // --mode=signal would tell the process set the value and wait for SIGINT or SIGTERM which needed when setting HIGH value 
+            string command = value == PinValue.High ? $"--mode=signal {GpioChip0} {pinNumber}=1" : $"{GpioChip0} {pinNumber}=0";
+
+            Process gpioSetCommand = new Process();
```

Comment:
Please add a `using` statement as Process is disposable. Otherwise you leak an OS handle to the finalizer thread.